### DISCUSSION TITLE
Possible new Baseline? (see link)

### DISCUSSION THREAD

**Discussion by Husky110** [AbdBarho/stable-diffusion-webui-docker#597] [Show and tell]
Hi there,  
Well - I know that the repo states "Start a discussion before you implement anything", but I kinda wanted the repo to be more what I expect it to be, so I implemented some stuff to make A1111 behave the way I want to use it.  
So I created a fork (see: https://github.com/Husky110/stable-diffusion-webui-docker)  and implemented A1111 in a way that fixes serveral issues, most importantly the file-permission-problems.  
I just wanted to ask for feedback on this and ask if it's viable to create a pull-request to maybe let this repo continue on my work (I will adapt ComfyUI aswell - haven't had much to do with Invoke so far...) or if it's better to start a new repo with this. My main-problem right now is that I am not sure if I can maintain my fork once I consider it "finished" (having a kid and beeing a freelancer tends to set timelimits ;) ). So I would really like for it to be continued. :)   
Thanks

**Comment by AbdBarho** [comment]
Hey!

Yes! contributions are welcome :)

I did not go through the all of the commits in your repo, could you please describe a bit more what kind of changes were done? I saw some more folders / models were added.

in any case, feel free to create a PR, I will review as soon as I can.

If you feel that the PR would be too big, feel free to divide into smaller ones.

**Reply by Husky110** [reply]
Good morning,

Of course I can add more detail. :)  
Most of the stuff is already in the Readme.md-file on the repo-frontpage, but the summary is this:
- all containers run as user 1000:1000 (gets rid of all the permission-problems on the host-system)
- most configs can be handled via an .env-file that is interpreted by docker
- installing and running are handled in custom shell-scripts
- A1111 comes with some recommended extensions preinstalled (Tiled VAE, Regional Prompter, ControlNet - can be disabled via -env-file)
- A1111 clears it's temp-directory when the container is started
- I've changed the directory-structure a bit so it is a bit more structured

In the end - for now the most important files are:
- docker-compose.yaml, install.sh, run.sh, services/download (whole directory), services AUTOMATIC1111/Dockerfile + entrypoint.sh

There is still room for improvement tho, but I wanna get ComfyUI running before I tackle that in general.

P.S.: Please don't mind the commit-wall - I am testing my stuff not on the machine I am developing on. :)

**Reply by DevilaN** [reply]
LOL. But my uid is 1001 in my system; also there are multiple accounts for my family that are using same docker image to run Stable Diffusion for themself and their uids are different from hardcoded 1000 too. I don't want to create new image multiple times for each user.
